libgloss/arc/hl/hl_gettimeofday.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * hl_gettimeofday.c -- provide _gettimeofday().
+ *
+ * Copyright (c) 2021 Synopsys Inc.
+ *
+ * The authors hereby grant permission to use, copy, modify, distribute,
+ * and license this software and its documentation for any purpose, provided
+ * that existing copyright notices are retained in all copies and that this
+ * notice is included verbatim in any distributions. No written agreement,
+ * license, or royalty fee is required for any of the authorized uses.
+ * Modifications to this software may be copyrighted by their authors
+ * and need not follow the licensing terms described here, provided that
+ * the new terms are clearly indicated on the first page of each file where
+ * they apply.
+ *
+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <sys/time.h>
+
+#include "hl_toolchain.h"
+#include "hl_api.h"
+
+/* Get host time(). Implements HL_SYSCALL_TIME. */
+static int
+hl_time (uint32_t *host_timer)
+{
+  int ret;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_TIME, ":i", (uint32_t *) host_timer);
+
+  if (p == NULL)
+    {
+      errno = ETIMEDOUT;
+      ret = -1;
+    }
+  else
+    {
+      ret = 0;
+    }
+
+  hl_delete ();
+
+  return ret;
+}
+
+/* gettimeofday() implementation. timezone and usec are not supported. */
+int
+_gettimeofday (struct timeval *tv, struct timezone *tz)
+{
+  int ret;
+  uint32_t host_timer;
+
+  if (tz != NULL)
```

Comment:
I think we should get user a chance to process an error here.

Reply:
We do not provide full functionality of `gettimeofday()`. Instead when the user calls this function with a `tz` struct, we just do nothing and throw a _timeout_ error. That is not clear to the user that why this happened. I'd say we should define a contract ("do not expect timezone data!") and if the user is in breach of that contract, they should know it by an abrupt abortion rather than a _timeout_ error that could have happened for whatever reason.

Nevertheless, I don't feel strong about this. So I am OK if you mark this as resolved.